src/aggregators/aggregators.jl

```diff
@@ -184,6 +184,38 @@ needs_vartojumps_map(aggregator::RSSACR) = true
 supports_variablerates(aggregator::AbstractAggregatorAlgorithm) = false
 supports_variablerates(aggregator::Coevolve) = true
 
+# true if aggregator supports hops, e.g. diffusion
 is_spatial(aggregator::AbstractAggregatorAlgorithm) = false
 is_spatial(aggregator::NSM) = true
 is_spatial(aggregator::DirectCRDirect) = true
+
+# return the fastest aggregator out of the available ones
+function select_aggregator(jumps::JumpSet; vartojumps_map=nothing, jumptovars_map=nothing, dep_graph=nothing, spatial_system=nothing, hopping_constants=nothing)
+    
+    # detect if a spatial SSA should be used
+    !isnothing(spatial_system) && !isnothing(hopping_constants) && return DirectCRDirect
+
+    # if variable rate jumps are present, return one of the two SSAs that support them
+    if num_vrjs(jumps) != 0
+        any(isbounded, vrjs) && return Coevolve
+        return Direct
+    end
+    
+    # if the number of jumps is small, return the Direct
+    num_jumps(jumps) < 10 && return Direct
+
+    # if there are only massaction jumps, we can any build dependency graph
```

Comment:
"we can any build dependency graph" 
Sounds like a weird sentence structure?

Reply:
I rewrote it as "we can any build the species-jumps dependency graphs".

Reply:
I still think "..., we can any build ..." sounds like strange English. I'm not a native speaker though (and it is only a comment), so no need to dwell on it.

Reply:
That was a genuine typo. I fixed it now.